Unable to compile

Unable to compile with MSBuild from .NET Framework 4.7.2 Developer Pack

C:\Users\Thomas>msbuild
Microsoft (R) Build Engine version 4.8.4084.0
[Microsoft .NET Framework, version 4.0.30319.42000]


There are many errors

`C:\Users\Thomas\Downloads\FolderSecurityViewer-main\FolderSecurityViewer-main\sourcecodes\FSV.Extensions.DependencyInje
ction\FSV.Extensions.DependencyInjection.csproj(1,1): error MSB4041: The default XML namespace of the project must be t
he MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.micro
soft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 forma
t, please convert it to MSBuild 2003 format.`

`"C:\Users\Thomas\Downloads\FolderSecurityViewer-main\FolderSecurityViewer-main\sourcecodes\FSV.FileSystem.Interop.Abstr
actions\FSV.FileSystem.Interop.Abstractions.csproj" (default target) (4:3) ->
  C:\Users\Thomas\Downloads\FolderSecurityViewer-main\FolderSecurityViewer-main\sourcecodes\FSV.FileSystem.Interop.Abst
ractions\FSV.FileSystem.Interop.Abstractions.csproj(1,1): error MSB4041: The default XML namespace of the project must
be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.m
icrosoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 f
ormat, please convert it to MSBuild 2003 format.`

`"C:\Users\Thomas\Downloads\FolderSecurityViewer-main\FolderSecurityViewer-main\sourcecodes\FSV.Resources\FSV.Resources.
csproj" (default target) (15) ->
(CoreCompile target) ->
  CSC : error CS1617: Invalid option 'latest' for /langversion; must be ISO-1, ISO-2, 3, 4, 5 or Default [C:\Users\Thom
as\Downloads\FolderSecurityViewer-main\FolderSecurityViewer-main\sourcecodes\FSV.Resources\FSV.Resources.csproj]`


At the end 

`    7 Warning(s)
    61 Error(s)`
`

--- a/README.md
+++ b/README.md
@@ -39,11 +39,13 @@ The application can be compiled on Windows, and depends on the following Framewo
 
 * .NET Framework 4.7.2
 * Optional: Visual Studio 2022
+* MSBuild version 17.9.8 or above
 
+### Build
 
 ````bash
 $ cd sourcecodes
-$ msbuild ./FolderSecurityViewer.sln /p:Configuration=Release
+$ msbuild ./FolderSecurityViewer.sln /t:restore /p:Configuration=Release
 ````
 
 

--- a/README.md
+++ b/README.md
@@ -37,7 +37,7 @@ If you want to automate your NTFS permissions reporting, you can use the Command
 
 The application can be compiled on Windows, and depends on the following Frameworks and utilities:
 
-* .NET Framework 4.7.2
+* .NET Framework 4.7.2, and .NET Framework 4.8
 * Optional: Visual Studio 2022
 * MSBuild version 17.9.8 or above
 

--- a/sourcecodes/FSV.Configuration.UnitTest/FSV.Configuration.UnitTest.csproj
+++ b/sourcecodes/FSV.Configuration.UnitTest/FSV.Configuration.UnitTest.csproj
@@ -19,6 +19,7 @@
     <TestProjectType>UnitTest</TestProjectType>
     <NuGetPackageImportStamp>
     </NuGetPackageImportStamp>
+    <LangVersion>default</LangVersion>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>

--- a/sourcecodes/FSV.Console.UnitTest/FSV.Console.UnitTest.csproj
+++ b/sourcecodes/FSV.Console.UnitTest/FSV.Console.UnitTest.csproj
@@ -9,7 +9,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>FSV.Console.UnitTest</RootNamespace>
     <AssemblyName>FSV.Console.UnitTest</AssemblyName>
-    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
     <ProjectTypeGuids>{3AC096D0-A1C2-E12C-1390-A8335801FDAB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
     <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">15.0</VisualStudioVersion>

--- a/sourcecodes/FSV.Crypto.UnitTest/FSV.Crypto.UnitTest.csproj
+++ b/sourcecodes/FSV.Crypto.UnitTest/FSV.Crypto.UnitTest.csproj
@@ -9,7 +9,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>FSV.Crypto.UnitTest</RootNamespace>
     <AssemblyName>FSV.Crypto.UnitTest</AssemblyName>
-    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
     <ProjectTypeGuids>{3AC096D0-A1C2-E12C-1390-A8335801FDAB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
     <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">15.0</VisualStudioVersion>

--- a/sourcecodes/FSV.Database.UnitTest/FSV.Database.UnitTest.csproj
+++ b/sourcecodes/FSV.Database.UnitTest/FSV.Database.UnitTest.csproj
@@ -19,6 +19,7 @@
     <TestProjectType>UnitTest</TestProjectType>
     <NuGetPackageImportStamp>
     </NuGetPackageImportStamp>
+    <LangVersion>default</LangVersion>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>

--- a/sourcecodes/FSV.Extensions.DependencyInjection.UnitTest/FSV.Extensions.DependencyInjection.UnitTest.csproj
+++ b/sourcecodes/FSV.Extensions.DependencyInjection.UnitTest/FSV.Extensions.DependencyInjection.UnitTest.csproj
@@ -1,9 +1,11 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFramework>netcoreapp3.1</TargetFramework>
+        <TargetFramework>net48</TargetFramework>
 
         <IsPackable>false</IsPackable>
+
+        <LangVersion>default</LangVersion>
     </PropertyGroup>
 
     <ItemGroup>

--- a/sourcecodes/FSV.Extensions.Serialization.UnitTest/FSV.Extensions.Serialization.UnitTest.csproj
+++ b/sourcecodes/FSV.Extensions.Serialization.UnitTest/FSV.Extensions.Serialization.UnitTest.csproj
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFramework>net5.0</TargetFramework>
+        <TargetFramework>net48</TargetFramework>
         <LangVersion>latest</LangVersion>
         <IsPackable>false</IsPackable>
     </PropertyGroup>

--- a/sourcecodes/FSV.Extensions.Serialization.UnitTest/ImmutableAvenger.cs
+++ b/sourcecodes/FSV.Extensions.Serialization.UnitTest/ImmutableAvenger.cs
@@ -1,5 +1,5 @@
 // FolderSecurityViewer is an easy-to-use NTFS permissions tool that helps you effectively trace down all security owners of your data.
-// Copyright (C) 2015 - 2024  Carsten Sch�fer, Matthias Friedrich, and Ritesh Gite
+// Copyright (C) 2015 - 2024  Carsten Schäfer, Matthias Friedrich, and Ritesh Gite
 // 
 // This program is free software: you can redistribute it and/or modify
 // it under the terms of the GNU Affero General Public License as

--- a/sourcecodes/FSV.Extensions.Serialization.UnitTest/ImmutableAvenger.cs
+++ b/sourcecodes/FSV.Extensions.Serialization.UnitTest/ImmutableAvenger.cs
@@ -1,5 +1,5 @@
 // FolderSecurityViewer is an easy-to-use NTFS permissions tool that helps you effectively trace down all security owners of your data.
-// Copyright (C) 2015 - 2024  Carsten Schäfer, Matthias Friedrich, and Ritesh Gite
+// Copyright (C) 2015 - 2024  Carsten Schï¿½fer, Matthias Friedrich, and Ritesh Gite
 // 
 // This program is free software: you can redistribute it and/or modify
 // it under the terms of the GNU Affero General Public License as
@@ -31,6 +31,7 @@ namespace FSV.Extensions.Serialization.UnitTest
         public string Name { get; }
         public string AlterEgo { get; }
 
+
         public bool Equals(ImmutableAvenger other)
         {
             if (ReferenceEquals(null, other))
@@ -48,12 +49,30 @@ namespace FSV.Extensions.Serialization.UnitTest
 
         public override bool Equals(object obj)
         {
-            return this.Equals(obj as ImmutableAvenger);
+            if (ReferenceEquals(null, obj))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, obj))
+            {
+                return true;
+            }
+
+            if (obj.GetType() != this.GetType())
+            {
+                return false;
+            }
+
+            return Equals((ImmutableAvenger)obj);
         }
 
         public override int GetHashCode()
         {
-            return HashCode.Combine(this.Name, this.AlterEgo);
+            unchecked
+            {
+                return ((this.Name != null ? this.Name.GetHashCode() : 0) * 397) ^ (this.AlterEgo != null ? this.AlterEgo.GetHashCode() : 0);
+            }
         }
     }
 }

--- a/sourcecodes/FSV.Extensions.Serialization.UnitTest/JsonSerializationWrapperTest.cs
+++ b/sourcecodes/FSV.Extensions.Serialization.UnitTest/JsonSerializationWrapperTest.cs
@@ -1,5 +1,5 @@
 // FolderSecurityViewer is an easy-to-use NTFS permissions tool that helps you effectively trace down all security owners of your data.
-// Copyright (C) 2015 - 2024  Carsten Sch�fer, Matthias Friedrich, and Ritesh Gite
+// Copyright (C) 2015 - 2024  Carsten Schäfer, Matthias Friedrich, and Ritesh Gite
 // 
 // This program is free software: you can redistribute it and/or modify
 // it under the terms of the GNU Affero General Public License as

--- a/sourcecodes/FSV.Extensions.Serialization.UnitTest/JsonSerializationWrapperTest.cs
+++ b/sourcecodes/FSV.Extensions.Serialization.UnitTest/JsonSerializationWrapperTest.cs
@@ -1,5 +1,5 @@
 // FolderSecurityViewer is an easy-to-use NTFS permissions tool that helps you effectively trace down all security owners of your data.
-// Copyright (C) 2015 - 2024  Carsten Schäfer, Matthias Friedrich, and Ritesh Gite
+// Copyright (C) 2015 - 2024  Carsten Schï¿½fer, Matthias Friedrich, and Ritesh Gite
 // 
 // This program is free software: you can redistribute it and/or modify
 // it under the terms of the GNU Affero General Public License as
@@ -60,7 +60,7 @@ namespace FSV.Extensions.Serialization.UnitTest
 
             ISerializationWrapper sut = wrapperFactory(SerializerType.Json);
 
-            await using var stream = new MemoryStream();
+            using var stream = new MemoryStream();
 
             // Act
             await sut.SerializeAsync(stream, expected);
@@ -87,7 +87,7 @@ namespace FSV.Extensions.Serialization.UnitTest
             string json = JsonConvert.SerializeObject(model);
             byte[] bytes = Encoding.UTF8.GetBytes(json);
 
-            target.Write(bytes);
+            target.Write(bytes, 0, bytes.Length);
             target.Seek(0, SeekOrigin.Begin);
         }
     }

--- a/sourcecodes/FSV.Extensions.Serialization.UnitTest/MutableAvenger.cs
+++ b/sourcecodes/FSV.Extensions.Serialization.UnitTest/MutableAvenger.cs
@@ -1,5 +1,5 @@
 // FolderSecurityViewer is an easy-to-use NTFS permissions tool that helps you effectively trace down all security owners of your data.
-// Copyright (C) 2015 - 2024  Carsten Sch�fer, Matthias Friedrich, and Ritesh Gite
+// Copyright (C) 2015 - 2024  Carsten Schäfer, Matthias Friedrich, and Ritesh Gite
 // 
 // This program is free software: you can redistribute it and/or modify
 // it under the terms of the GNU Affero General Public License as

--- a/sourcecodes/FSV.Extensions.Serialization.UnitTest/MutableAvenger.cs
+++ b/sourcecodes/FSV.Extensions.Serialization.UnitTest/MutableAvenger.cs
@@ -1,5 +1,5 @@
 // FolderSecurityViewer is an easy-to-use NTFS permissions tool that helps you effectively trace down all security owners of your data.
-// Copyright (C) 2015 - 2024  Carsten Schäfer, Matthias Friedrich, and Ritesh Gite
+// Copyright (C) 2015 - 2024  Carsten Schï¿½fer, Matthias Friedrich, and Ritesh Gite
 // 
 // This program is free software: you can redistribute it and/or modify
 // it under the terms of the GNU Affero General Public License as
@@ -57,12 +57,15 @@ namespace FSV.Extensions.Serialization.UnitTest
                 return false;
             }
 
-            return this.Equals((MutableAvenger)obj);
+            return Equals((MutableAvenger)obj);
         }
 
         public override int GetHashCode()
         {
-            return HashCode.Combine(this.Name, this.AlterEgo);
+            unchecked
+            {
+                return ((this.Name != null ? this.Name.GetHashCode() : 0) * 397) ^ (this.AlterEgo != null ? this.AlterEgo.GetHashCode() : 0);
+            }
         }
     }
 }

--- a/sourcecodes/FSV.Extensions.WindowConfiguration.Test/FSV.Extensions.WindowConfiguration.Test.csproj
+++ b/sourcecodes/FSV.Extensions.WindowConfiguration.Test/FSV.Extensions.WindowConfiguration.Test.csproj
@@ -9,7 +9,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>FSV.Extensions.WindowConfiguration.Test</RootNamespace>
     <AssemblyName>FSV.Extensions.WindowConfiguration.Test</AssemblyName>
-    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
     <ProjectTypeGuids>{3AC096D0-A1C2-E12C-1390-A8335801FDAB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
     <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">15.0</VisualStudioVersion>

--- a/sourcecodes/FSV.FileSystem.Interop.UnitTest/FSV.FileSystem.Interop.UnitTest.csproj
+++ b/sourcecodes/FSV.FileSystem.Interop.UnitTest/FSV.FileSystem.Interop.UnitTest.csproj
@@ -9,7 +9,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>FSV.FileSystem.Interop.UnitTest</RootNamespace>
     <AssemblyName>FSV.FileSystem.Interop.UnitTest</AssemblyName>
-    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
     <ProjectTypeGuids>{3AC096D0-A1C2-E12C-1390-A8335801FDAB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
     <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">15.0</VisualStudioVersion>

--- a/sourcecodes/FSV.ShareServices.UnitTest/FSV.ShareServices.UnitTest.csproj
+++ b/sourcecodes/FSV.ShareServices.UnitTest/FSV.ShareServices.UnitTest.csproj
@@ -9,7 +9,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>FSV.ShareServices.UnitTest</RootNamespace>
     <AssemblyName>FSV.ShareServices.UnitTest</AssemblyName>
-    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
     <ProjectTypeGuids>{3AC096D0-A1C2-E12C-1390-A8335801FDAB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
     <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">15.0</VisualStudioVersion>
@@ -19,6 +19,7 @@
     <TestProjectType>UnitTest</TestProjectType>
     <NuGetPackageImportStamp>
     </NuGetPackageImportStamp>
+    <LangVersion>default</LangVersion>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>

--- a/sourcecodes/FSV.ViewModel.UnitTest/app.config
+++ b/sourcecodes/FSV.ViewModel.UnitTest/app.config
@@ -11,7 +11,7 @@
             <dependentAssembly>
                 <assemblyIdentity name="System.Runtime.CompilerServices.Unsafe" publicKeyToken="b03f5f7f11d50a3a"
                                   culture="neutral" />
-                <bindingRedirect oldVersion="0.0.0.0-5.0.0.0" newVersion="5.0.0.0" />
+                <bindingRedirect oldVersion="0.0.0.0-6.0.0.0" newVersion="6.0.0.0" />
             </dependentAssembly>
         </assemblyBinding>
     </runtime>

--- a/sourcecodes/FolderSecurityViewer/FolderSecurityViewer.csproj
+++ b/sourcecodes/FolderSecurityViewer/FolderSecurityViewer.csproj
@@ -33,6 +33,7 @@
     <ApplicationVersion>1.0.0.%2a</ApplicationVersion>
     <UseApplicationTrust>false</UseApplicationTrust>
     <BootstrapperEnabled>true</BootstrapperEnabled>
+    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <PlatformTarget>AnyCPU</PlatformTarget>
@@ -61,14 +62,6 @@
   </PropertyGroup>
   <PropertyGroup>
     <ApplicationIcon>Resources\Images\AppIcon.ico</ApplicationIcon>
-  </PropertyGroup>
-  <PropertyGroup>
-    <Win32Resource>
-    </Win32Resource>
-  </PropertyGroup>
-  <PropertyGroup />
-  <PropertyGroup>
-    <SignAssembly>true</SignAssembly>
   </PropertyGroup>
   <ItemGroup>
     <Reference Include="System" />

